plugins/modules/dns_auth_nsg.py

```diff
@@ -0,0 +1,605 @@
+                            returned: Always
+                        protocol_name:
+                            description:
+                                - "TSIG key name in punycode."
+                            type: str
+                            returned: Always
+                        secret:
+                            description:
+                                - "TSIG key secret, base64 string."
+                            type: str
+                            returned: Always
+        id:
+            description:
+                - "The resource identifier."
+            type: str
+            returned: Always
+        internal_secondaries:
+            description:
+                - "Optional. BloxOne DDI hosts acting as internal secondaries. Order is not significant."
+            type: list
+            returned: Always
+            elements: dict
+            contains:
+                host:
+                    description:
+                        - "The resource identifier."
+                    type: str
+                    returned: Always
+        name:
+            description:
+                - "Name of the object."
+            type: str
+            returned: Always
+        nsgs:
+            description:
+                - "The resource identifier."
+            type: list
+            returned: Always
+        tags:
+            description:
+                - "Tagging specifics."
+            type: dict
+            returned: Always
+"""  # noqa: E501
+
+from ansible_collections.infoblox.bloxone.plugins.module_utils.modules import BloxoneAnsibleModule
+
+try:
+    from bloxone_client import ApiException, NotFoundException
+    from dns_config import AuthNSG, AuthNsgApi
+except ImportError:
+    pass  # Handled by BloxoneAnsibleModule
+
+
+class AuthNsgModule(BloxoneAnsibleModule):
+    def __init__(self, *args, **kwargs):
+        super(AuthNsgModule, self).__init__(*args, **kwargs)
+
+        exclude = ["state", "csp_url", "api_key", "id"]
+        self._payload_params = {k: v for k, v in self.params.items() if v is not None and k not in exclude}
+        self._payload = AuthNSG.from_dict(self._payload_params)
+        self._existing = None
+
+    @property
+    def existing(self):
+        return self._existing
+
+    @existing.setter
+    def existing(self, value):
+        self._existing = value
+
+    @property
+    def payload_params(self):
+        return self._payload_params
+
+    @property
+    def payload(self):
+        return self._payload
+
+    def payload_changed(self):
+        if self.existing is None:
+            # if existing is None, then it is a create operation
+            return True
+
+        return self.is_changed(self.existing.model_dump(by_alias=True, exclude_none=True), self.payload_params)
+
+    def find(self):
+        if self.params["id"] is not None:
+            try:
+                resp = AuthNsgApi(self.client).read(self.params["id"], inherit="full")
```

Comment:
remove inherit param from the file if not supported

Reply:
done